CHANGELOG.md

```diff
@@ -9,6 +9,10 @@
   - `array_flatten`
 - Added support for replicating your local Python environment on Snowflake via `Session.replicate_local_environment`.
 
+### Behavior Changes
+
+- When creating stored procedures, UDFs, UDTFs, UDAFs with parameter `is_permanent=False` will now create temporary objects even when `stage_name` is provided. Earlier `is_permanent=False` with non-None `stage_name` try to create permanent objects at the given stage location.
```

Comment:
I think we also need to call out that the default behavior is changing because the default value of `is_permanent` is `None`. So the behavior of a UDF that only has `stage_name` specified will see their UDF becomes temporary.